girder/models/folder.py

```diff
@@ -735,8 +735,13 @@ def copyFolder(self, srcFolder, parent=None, name=None, description=None,
             name = srcFolder['name']
         if description is None:
             description = srcFolder['description']
-        if public == 'original':
-            public = srcFolder.get('public', None)
+        if public is not None and isinstance(public, six.string_types):
+            if public == 'original':
+                public = srcFolder.get('public', None)
+            elif public.lower() == 'true':
```

Comment:
```
else:
    public = self.boolParam('public', params)
```

In any case, the `lower()` is not necessary since `autoDescribeRoute` will validate that a valid value is passed from the enum.